pandas/core/ops/__init__.py

```diff
@@ -705,6 +705,52 @@ def to_series(right):
     return left, right
 
 
+def _should_reindex_op(left, right, axis, default_axis, fill_value, level) -> bool:
+    """
+    Check if this is an operation between DataFrames that will need to reindex.
+    """
+    if not isinstance(right, ABCDataFrame):
+        return False
+
+    if fill_value is None and level is None and axis is default_axis:
+        # TODO: any other cases we should handle here?
+        cols = left.columns.intersection(right.columns)
```

Comment:
isn't this just
set(left.columns) == set(right.columns) ?

Reply:
?

Reply:
yah, but for certain types of indexes (e.g. RangeIndex) intersection is optimized.  Usually won't be enough to matter, but still

Reply:
Do you know if `xor` also optimized? I think this is equivalent to `not len(left.columns ^ right.columns)`. My guess is that xor will tend to be slower than your check.

Reply:
ok, fair point. yeah we likley just need some more asv's around this (followup)